### PR TITLE
Fix build after upgrading to SB 2.2

### PR DESCRIPTION
**Pull request by meltsufin**
It appears that the `Envirnment` bean is not available anymore in the context during bootstrap configuration. So, switching `GcpConfigProperties` to `ApplicationContextAware`. Also, fixing the fact that if config module is disabled, we shouldn't even have an instance of `GcpConfigProperties` bean.

### CODE REVIEW

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java [hunk 1]
```diff
@@ -130,4 +116,20 @@ public Credentials getCredentials() {
 		return this.credentials;
 	}
 
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		if (this.profile == null) {
+			String[] profiles = applicationContext.getEnvironment().getActiveProfiles();
```
Comment (elefeint) [inline spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java]: Should we add `Assert.notNull()` on `applicationContext` and` applicationContext.getEnvironment()`, to simplify troubleshooting in the future?

File: spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigBootstrapConfigurationTest.java [hunk 2]
```diff
@@ -51,15 +54,23 @@ public void testConfigurationValuesAreCorrectlyLoaded() {
...
-					assertThat(config.isEnabled()).isFalse();
+					assertThat(config.isEnabled()).isTrue();
 					assertThat(config.getProjectId()).isEqualTo("pariah");
 				});
 	}
+
+	@Test
+	public void testConfigurationDisabled() {
+		this.contextRunner.run((context) -> {
+					assertThatThrownBy(() -> context.getBean(GcpConfigProperties.class)).isInstanceOf(
```
Comment (elefeint) [inline spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigBootstrapConfigurationTest.java]: Line wrapping is a bit odd on this.

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java (hunk 1 above)
Comment (meltsufin) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java]: I believe the contract of `ApplicationContextAware` is that it provides a non-null context, and the context should always have an environment.

**Comment by meltsufin** [comment]
It's odd that it fails on Travis because `mvn clean install' passed for me locally.

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java (hunk 1 above)
Comment (artembilan) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java]: That's correct.
Although I think we can narrow the responsibility to the `EnvironmentAware`  hook: nor reason to deal with the whole `ApplicationContext` here.

File: spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigBootstrapConfigurationTest.java [hunk 3]
```diff
@@ -51,15 +54,23 @@ public void testConfigurationValuesAreCorrectlyLoaded() {
...
-					assertThat(config.isEnabled()).isFalse();
+					assertThat(config.isEnabled()).isTrue();
 					assertThat(config.getProjectId()).isEqualTo("pariah");
 				});
 	}
+
+	@Test
+	public void testConfigurationDisabled() {
+		this.contextRunner.run((context) -> {
+			assertThatThrownBy(() -> context.getBean(GcpConfigProperties.class)).isInstanceOf(
```
Comment (artembilan) [inline spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigBootstrapConfigurationTest.java]: Not related and minor, but we recently opened for ourselves that `assertThatExceptionOfType()` is much handy for this kind of verifications: https://github.com/spring-projects/spring-integration/issues/2760

**Comment by meltsufin** [comment]
Getting a login screen in `GcpStorageAutoConfigurationTests` and just can't figure out which auto configuration to exclude.

**Comment by meltsufin** [comment]
Merging, but adding #1521 to be worked on to unignore the failing test.

**Pull request merged**